### DISCUSSION TITLE
Invalid configuration on Edge 830

### DISCUSSION THREAD

**Discussion by Sdb56011** [maca88/SmartBikeLights#18] [Q&A]
This did work flawlessly on my old Edge 830. That was just replaced under warranty. I just installed Smart Bike Lights again, made a new configuration, but any configuration at all is giving me Error 4 (invalid configuration). With none, I have all the fields, just labeled with random numbers, but as soon as I enter any configuration at all, it breaks. Any help or insights that anyone can offer would be appreciated, I really liked this screen

using:
Edge 830
Software v8.20
Bontrager Ion Pro RT
Bontrager Flare RT

the configurations generated by the tool which aren’t working (these are the two I would like, though I’ve tried any number of combinations, trying to isolate the problem):

#4587520,196641#8,4|Day:2:63:1Er600,s-600H]0|NightF:3:2:1H]0Es-600,r600C]6.7056|NightS:2:62:1Es-600,r600H]0|:1:0:1D=1#6291461,1409482753#4,3|Day:2:8:1Er600,s-600H]0|Night:1:63:1Es-600,r600|:1:0:1D=1#7,6:Ion Pro RT|2,:-1,Off:0|1,High:1|1,Medium:2|1,Low:5|1,Day Flash:63|1,Night Flash:62#7,6:Flare RT|2,:-1,Off:0|1,Day Steady:1|1,Night Steady:5|1,Day Flash:7|1,All-Day Flash:8|1,Night Flash:63#0::#0:0#B3122#1#1#1#0

#4587520,196641#1,1|:1:0:1D=1#6291461,1409482753#1,1|:1:0:1D=1#7,6:Ion Pro RT|2,:-1,Off:0|1,High:1|1,Medium:2|1,Low:5|1,Day Flash:63|1,Night Flash:62#7,6:Flare RT|2,:-1,Off:0|1,Day Steady:1|1,Night Steady:5|1,Day Flash:7|1,All-Day Flash:8|1,Night Flash:63#0::#0:0#B3122#1#1#0#0

**Comment by maca88** [comment]
Hi,

I tried to reproduce the issue with Edge 1000 + Ion Pro RT + Flare RT, but I was not able to get error 4, both configurations works fine. I also tried to reinstall the data field to see if a fresh install would be an issue, but it wasn't. The data field is compiled with the same flags for Edge 1000 and 830 (same code), so it is very strange why on my Edge it works but not on yours.

What did you use to install the data field, Garmin Express (Windows or OSX) or Garmin Connect Mobile (iOS or Android)?
In which application setting did you put the configuration ("Lights Configuration" or "Secondary/Tertiary Lights Configuration")?

**Reply by Sdb56011** [reply]
I installed using the Garmin Connect and Garming IQ apps, I am entering the configuration in the settings menu within the Garmin IQ app. I tried both the Light Configuration and Secondary Lights Configuration fields, changing the "Current Configuration" field accordingly

**Reply by maca88** [reply]
Unfortunately I am currently out of ideas what could be wrong, your are the second person with Edge 830 that is experiencing such issue. The [first person reported](https://forums.garmin.com/developer/connect-iq/f/showcase/248492/smart-bike-lights-data-field-bike-lights-control-widget/1333369#1333369) that it works only for one configuration bot not when having two, so it is not exactly the same. The only thing that I can do is to create a special version of the data field with added logging, which you would need to install on your Edge and then with the logs I would hopefully detect the issue. In case you are interested in helping solving the issue by trying a diagnostic version of the data field let me know, I will then prepare the version for you to try.

**Reply by maca88** [reply]
Based on the input from other users I was able to find the issue and fix it. It turned out that with the latest version of Connect IQ Store 2.12, they removed support for some characters, one of them was used by this data field (character |). The issue is fixed with the latest version 0.15.1, but you need to update your configuration by using the Light Configurator so that the character | is replaced to !.
Here are the updated configurations from above:

#4587520,196641#8,4!Day:2:63:1Er600,s-600H]0!NightF:3:2:1H]0Es-600,r600C]6.7056!NightS:2:62:1Es-600,r600H]0!:1:0:1D=1#6291461,1409482753#4,3!Day:2:8:1Er600,s-600H]0!Night:1:63:1Es-600,r600!:1:0:1D=1#7,6:Ion Pro RT!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62#7,6:Flare RT!2,:-1,Off:0!1,Day Steady:1!1,Night Steady:5!1,Day Flash:7!1,All-Day Flash:8!1,Night Flash:63#0::#0:0#B3122#1#1#1#0

#4587520,196641#1,1!:1:0:1D=1#6291461,1409482753#1,1!:1:0:1D=1#7,6:Ion Pro RT!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62#7,6:Flare RT!2,:-1,Off:0!1,Day Steady:1!1,Night Steady:5!1,Day Flash:7!1,All-Day Flash:8!1,Night Flash:63#0::#0:0#B3122#1#1#0#0

**Reply by Sdb56011** [reply]
Oh thank you! I just checked and everything is working again! Much appreciated, this app is really far better than Bontrager’s or Garmin’s native light control :)

**Comment by MirekZoubek** [comment]
Hello @maca88, I am having the same problem with 830 + Varia 515. Any but default (empty) configuration ends up with Error 4. No character | in it though...

Example: ###0,73535488:1,1193950855#1,1!:1:7:1D=1##6,5:Varia 515!2,:-1,Off:0!1,Solid:4!1,Peloton:5!1,Day Flash:7!1,Night Flash:6#0::#0:0#B3122##3#0#0

Any hints please?

**Reply by maca88** [reply]
Hi,

I tried your configuration on the simulator for Edge 830 and on a real Edge 1000, works fine on both.
Do you have the latest Smart Bike Lights installed (0.15.2)?
Which software do you use to set the configuration (Connect IQ Store, Garmin Connect Mobile, Garmin Express)?

**Reply by MirekZoubek** [reply]
Hi, I guess I need more clarification. There seem to be two things:

- Data Field - Smart Bike Lights (latest version is 0.15.2)
- Widget - Bike Lights Control (latest version is 0.8.2)

I need to install them both, right?

Only then the additional widget appears on my Edge 830 as seen in the picture below.

![image](https://user-images.githubusercontent.com/97151314/148277156-4a351083-869b-4fc3-8b63-71eceeb5a0ac.png)

The buttons work and Varia changes modes nicely.

If I then want to configure the buttons and behaviour differently, I insert the code (into the widget settings or into the data field settings? Widget setting gives me Error 4, data field settings does nothing.

I am sure I m doing something wrong, juts not sure what it is.

Thank you

**Reply by MirekZoubek** [reply]
And I forgot to say that I am using Garmin Connect Mobile to do all of the above.

**Reply by maca88** [reply]
You don't need to install both, you can use just Smart Bike Lights or Bike Lights Control.

Regarding the configuration, you need to be aware that the configuration for Smart Bike Lights will not work for Bike Lights Control, I am assuming that you used the configuration above for Bike Lights Control instead for Smart Bike Lights and that why you got Error 4.

After installing Smart Bike Lights data field, you need to place it in one of the data screens, you can check the following video how to do that: https://www.youtube.com/watch?v=7r2W483URk4

 For configuring the Bike Lights Control widget, you need to use this configurator: https://maca88.github.io/BikeLightsControl/, this configuration should work for you:

`###0,73535488###6,5:Varia 515!2,:-1,Off:0!1,Solid:4!1,Peloton:5!1,Day Flash:7!1,Night Flash:6#0::#0:0#B3122##3#0#0`

**Reply by MirekZoubek** [reply]
It is working now! Thanks for the clarification. I find the data field easier to access when cycling.
